R/modspec.R

```diff
@@ -196,24 +196,49 @@ check_spec_contents <-  function(x, crump = TRUE, warn = TRUE, ...) {
   return(invisible(NULL))
 }
 
-audit_spec <- function(x, spec, warn = TRUE) {
-  cmt <- names(init(x))
-  if(!has_name("ODE", spec) | !warn | length(cmt) ==0) {
+grepl_dxdt_ode <- function(spec, cmt) {
+  dx <- paste0("dxdt_", cmt)
+  z <- vapply(dx, FUN.VALUE = TRUE, function(dxi) {
+    pat <- paste0(" *\\Q", dxi, "\\E", " *=")
```

Comment:
Looking at my [suggestion](https://github.com/metrumresearchgroup/mrgsolve/pull/1235#discussion_r1779161968) again, that leading " *" is superfluous.  I'm guessing I meant to add a "^" anchor there, but that would have been a bad idea because, at the very least, things like "$ODE dxdt_a = 3" should match.

As is, the pattern leaves open the possibility of a partial match at the front end (e.g., "bahdxdt_CMT =").  I'm not sure how much of a worry that is, but my updated suggestion is

```diff
diff --git a/R/modspec.R b/R/modspec.R
index 27be48de..88ad80ef 100644
--- a/R/modspec.R
+++ b/R/modspec.R
@@ -199,7 +199,7 @@ check_spec_contents <-  function(x, crump = TRUE, warn = TRUE, ...) {
 grepl_dxdt_ode <- function(spec, cmt) {
   dx <- paste0("dxdt_", cmt)
   z <- vapply(dx, FUN.VALUE = TRUE, function(dxi) {
-    pat <- paste0(" *\\Q", dxi, "\\E", " *=")
+    pat <- paste0("\\b\\Q", dxi, "\\E", " *=")
     any(grepl(pat, spec[["ODE"]]))
   })  
   names(z) <- dx
```

Reply:
@kyleam Thanks; updated.